session_test.go

```diff
@@ -799,3 +805,58 @@ func TestBacklogExceeded_Accept(t *testing.T) {
 		}
 	}
 }
+
+func TestWindowUpdateWriteDuringRead(t *testing.T) {
+	client, server := testClientServer()
+	defer client.Close()
+	defer server.Close()
+
+	wg := &sync.WaitGroup{}
```

Comment:
An unimportant nitpick of style: if you're not passing a waitgroup around as a paramter, you can just use `var wg sync.WaitGroup`. Go automatically will turn the receiver into a pointer for function calls, and I think its prettier. But like I said, no functional difference.